### ISSUE TITLE
ClusterPreferredOwner: Set-TargetResource does not handle Ensure correctly

### ISSUE THREAD

**Issue by johlju** [dsccommunity/xFailOverCluster#97]
**Details of the scenario you tried and the problem that is occurring:**
Set-TargetResource will set preferred owners regardless if they are already correct or not. Set-TargetResource should only set preferred owner if they are not in desired state.

**The DSC configuration that is using the resource (as detailed as possible):**
n/a

**Version of the Operating System and PowerShell the DSC Target Node is running:**
n/a

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**
Dev

**Comment by fculpo** [comment]
Indeed, using this resource always returns a change with multiple consequent runs
